azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_provider.py

```diff
@@ -128,10 +137,27 @@ def disconnect(self):
 
     def publish(self, topic, message_payload):
         """
-        This method enables the transport to send a message to the message broker
+        This method enables the transport to send a message to the message broker.
+        By default the the quality of service level to use is set to 1.
         :param topic: topic: The topic that the message should be published on.
         :param message_payload: The actual message to send.
+        :return message ID for the publish request.
         """
         logger.info("sending")
         message_info = self._mqtt_client.publish(topic=topic, payload=message_payload, qos=1)
         return message_info.mid
+
+    def subscribe(self, topics, qos=0):
+        """
+        This method subscribes the client to one or more topics.
+        :param topics: a single string specifying the subscription topic to subscribe to
+        OR a single tuple of (topic,qos) or a list of tuples in the format of (topic,qos)
```

Comment:
What's the behavior of this function if you pass a list of tuples and it succeeds for some and fails for others?  I assume that we're inheriting this behavior from Paho, but if we're putting the same behavior on our object, we should define it.  We won't ever pass a tuple right now, but we'll need to subscribe to multiple topics for twin.

Reply:
Good point. easier to start with a simple "only one topic at a time" approach and make it smart for multiple topics later maybe?

---
In reply to: [246543480](https://github.com/Azure/azure-iot-sdk-python-preview/pull/36#discussion_r246543480) [](ancestors = 246543480)

Reply:
changed it to single topic but will continue to see what happens when its multiple

---
In reply to: [246898773](https://github.com/Azure/azure-iot-sdk-python-preview/pull/36#discussion_r246898773) [](ancestors = 246898773,246543480)